modules/caddyhttp/metrics.go

```diff
@@ -101,23 +103,29 @@ func serverNameFromContext(ctx context.Context) string {
 type metricsInstrumentedHandler struct {
 	handler string
 	mh      MiddlewareHandler
+	perHost bool
 }
 
-func newMetricsInstrumentedHandler(handler string, mh MiddlewareHandler) *metricsInstrumentedHandler {
+func newMetricsInstrumentedHandler(handler string, mh MiddlewareHandler, perHost bool) *metricsInstrumentedHandler {
 	httpMetrics.init.Do(func() {
 		initHTTPMetrics()
 	})
 
-	return &metricsInstrumentedHandler{handler, mh}
+	return &metricsInstrumentedHandler{handler, mh, perHost}
 }
 
 func (h *metricsInstrumentedHandler) ServeHTTP(w http.ResponseWriter, r *http.Request, next Handler) error {
+	host := "*"
+	if h.perHost {
+		host = r.Host
+	}
+
 	server := serverNameFromContext(r.Context())
-	labels := prometheus.Labels{"server": server, "handler": h.handler}
+	labels := prometheus.Labels{"server": server, "handler": h.handler, "host": host}
```

Comment:
should conditionally set `host`:

```suggestion
	labels := prometheus.Labels{"server": server, "handler": h.handler}
	if h.perHost {
		labels["host"] = host
	}
```